src/core/sorted_map.cc

```diff
@@ -309,9 +309,9 @@ optional<unsigned> SortedMap::GetRank(sds ele, bool reverse) const {
   if (obj == nullptr)
     return std::nullopt;
 
-  optional rank = score_tree->GetRank(obj);
+  optional rank = score_tree->GetRank(obj, reverse);
   DCHECK(rank);
-  return reverse ? score_map->UpperBoundSize() - *rank - 1 : *rank;
```

Comment:
why was UpperBoundSize() used here? It isn't obvious, perhaps something slipped away from me.

IMHO: I'd like to keep DCHECK. It's true that score_map->GetRank shouldn't return std::nullopt in this codepath. It can do it in general though. So it wouldn't hurt.

Reply:
dense_set has the ability to expire items but it does not do it proactively, so its size method may include items that are already expired. We do not use the feature in sorted set though.